backend/app/services/langgraph_chat/tools/skills/file_reader.py

```
@@ -0,0 +1,239 @@
+
+        Returns:
+            Absolute directory path
+
+        Raises:
+            ValueError: If path is outside workspace root
+        """
+        directory = directory.lstrip("/")
+        full_dir = os.path.abspath(os.path.join(self.workspace_root, directory))
+
+        if not full_dir.startswith(os.path.abspath(self.workspace_root)):
+            raise ValueError(f"Path traversal detected: {directory}")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)